core/src/packet_deserializer.rs

```diff
@@ -45,26 +45,36 @@ impl PacketDeserializer {
     ) -> Result<ReceivePacketResults, RecvTimeoutError> {
         let (packet_batches, sigverify_tracer_stats_option) =
             self.receive_until(recv_timeout, capacity)?;
+        Ok(Self::deserialize_and_collect_packets(
+            &packet_batches,
+            sigverify_tracer_stats_option,
+        ))
+    }
 
+    /// Deserialize packet batches and collect them into ReceivePacketResults
+    fn deserialize_and_collect_packets(
```

Comment:
nit: add tests for this function, as the PR's description suggests

Reply:
Ah, in the PR here I meant to make it easier to use in tests inside #27611.
I'll create another PR to add some testing here though.